### DISCUSSION TITLE
Enhanced depends `.Daemoned` result

### DISCUSSION THREAD

**Discussion by hippolytej** [argoproj/argo-workflows#10363] [Q&A] [answered]
Hi all,

First of all, I'm not sure I understand what is meant by a "Daemoned" task in the [Enhanced Depends Logic](https://argoproj.github.io/argo-workflows/enhanced-depends-logic/#depends) documentation, and would really appreciate it if someone could give some precisions on it 🙏

The reason I'm interested in this is that I'm wondering if there is a way to use the `depends` keyword to create a dag where two tasks are running simultaneously, say `task_A` and `task_B`, but one (`task_B`) would only get scheduled once the other (`task_A`) starts running. In other words, I'd like `task_B` not to be scheduled if for some reason (e.g. pod get stuck initializing, cannot find resources, etc.)  `task_A` never starts.

Thank you for your help!

**Comment by agilgur5** [comment, marked as answer]
> I'm not sure I understand what is meant by a "Daemoned" task in the [Enhanced Depends Logic](https://argoproj.github.io/argo-workflows/enhanced-depends-logic/#depends) documentation

It's referring to the [Daemon Containers](https://argo-workflows.readthedocs.io/en/latest/walk-through/daemon-containers/) feature. I've added a link to it from this page in #13110

> but one (`task_B`) would only get scheduled once the other (`task_A`) starts running

You can't do this with `depends` since it relies on the _results_ of a task, which in this case does not yet exist as it is still running.

What you could use to achieve this is template-level [hooks](https://argo-workflows.readthedocs.io/en/latest/lifecyclehook/), such as in [this example](https://github.com/argoproj/argo-workflows/blob/dc6a18dc545bbf35103ffddb81a014eefcbb403d/examples/life-cycle-hooks-tmpl-level.yaml#L14).
